### PR TITLE
Some basic documentation for arrays

### PR DESCRIPTION
**Pull request by b-mehta**
Add some documentation for destination arrays and polarized arrays.

Also remove some unnecessary pragmas and fix a loop in linear monoids.

### CODE REVIEW

File: src/Data/Array/Polarized/Push.hs [hunk 1]
```diff
@@ -19,6 +20,16 @@ import Prelude.Linear
 import qualified Prelude
 import Data.Monoid.Linear
 
+-- XXX: the below isn't really true yet since no friendly way of constructing a
+-- PushArray directly is given yet (see issue #62), but the spirit holds.
+-- XXX: There's also a slight lie in that one might one to consume a PushArray
```
Comment (monoidal) [inline src/Data/Array/Polarized/Push.hs]: might want

File: src/Data/Array/Destination.hs [hunk 2]
```diff
@@ -104,7 +105,9 @@ mirror as f
       fill (f x) dl `lseq` mirror xs f dr
 
 -- TODO: It would be better to implement this directly, instead of in terms of
--- mirror, to avoid allocation.
+-- mirror, to avoid allocation. This would also mean the length argument is
+-- redundant, as it should be.
+-- | Fill a destination array using the given function and length.
```
Comment (monoidal) [inline src/Data/Array/Destination.hs]: After #66, there's no length argument anymore.

File: src/Data/Array/Polarized/Push.hs (hunk 1 above)
Comment (b-mehta) [reply src/Data/Array/Polarized/Push.hs]: Agreed, I'll rewrite these to be TODOs - it's not yet clear what a good way of doing this is.

Comment (monoidal) [reply src/Data/Array/Polarized/Push.hs]: I meant "might one" -> "might want"

Comment (b-mehta) [reply src/Data/Array/Polarized/Push.hs]: Ah. My mistake, thanks for this catch.

**Pull request merged**